sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -409,6 +409,25 @@ class SessionCatalog(
   }
 
   /**
+   * Returns the partition names from catalog for a given table in a database.
+   */
+  def getPartitionNames(db: String, table: String, range: Short): Seq[String] = {
+    externalCatalog.getPartitionNames(db, table, range)
+  }
+
+  /**
+   * Returns the partition names that matche the partition spec for a given table in a database.
+   * When no match is found, an empty Sequence is returned.
+   */
+  def getPartitionNames(
```

Comment:
+1 we should follow current catalog API especially we already have `listPartitions` method.